src/platform/packages/shared/kbn-unified-metrics-grid/src/components/trace_metrics_grid/error_rate.tsx

```diff
@@ -0,0 +1,101 @@
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { evaluate, from, keep, sort, stats, where } from '@kbn/esql-composer';
+import type { ChartSectionProps, UnifiedHistogramInputMessage } from '@kbn/unified-histogram/types';
+import React, { useRef } from 'react';
+import type { Observable } from 'rxjs';
+import { AT_TIMESTAMP, EVENT_OUTCOME, PROCESSOR_EVENT, STATUS_CODE } from '@kbn/apm-types';
+import type { DataSource } from '.';
+import { chartPalette } from '.';
+import { useLensProps } from '../chart/hooks/use_lens_props';
+import { LensWrapper } from '../chart/lens_wrapper';
+import { ChartContainer } from '../chart_container';
+
+interface Props {
+  indexes: string;
+  getTimeRange: ChartSectionProps['requestParams']['getTimeRange'];
+  services: ChartSectionProps['services'];
+  discoverFetch$: Observable<UnifiedHistogramInputMessage>;
+  searchSessionId?: string;
+  abortController?: AbortController;
+  onBrushEnd: ChartSectionProps['onBrushEnd'];
+  onFilter: ChartSectionProps['onFilter'];
+  filters: string[];
+  dataSource: DataSource;
+}
+
+function getQuery(dataSource: DataSource, indexes: string, filters: string[]) {
+  const whereClauses = [...filters, `${PROCESSOR_EVENT} == "transaction"`].map((filter) =>
+    where(filter)
+  );
+  return from(indexes)
+    .pipe(
+      ...whereClauses,
+      dataSource === 'apm'
+        ? stats(
+            `failure = COUNT(*) WHERE ${EVENT_OUTCOME} == "failure", all = COUNT(*)  BY timestamp = BUCKET(${AT_TIMESTAMP}, 100, ?_tstart, ?_tend)`
+          )
+        : stats(
+            `failure = COUNT(*) WHERE ${STATUS_CODE} == "Error", all = COUNT(*)  BY timestamp = BUCKET(${AT_TIMESTAMP}, 100, ?_tstart, ?_tend)`
+          ),
+      evaluate('error_rate = TO_DOUBLE(failure) / all'),
+      keep('timestamp, error_rate'),
+      sort('timestamp')
+    )
+    .toString();
+}
+
+export const ErrorRateChart = ({
+  indexes,
+  getTimeRange,
+  services,
+  discoverFetch$,
+  searchSessionId,
+  abortController,
+  onBrushEnd,
+  onFilter,
+  filters,
+  dataSource,
+}: Props) => {
+  const chartRef = useRef<HTMLDivElement>(null);
+  const query = getQuery(dataSource, indexes, filters);
+
+  const errorRateLensProps = useLensProps({
+    title: 'Error Rate',
+    query,
+    getTimeRange,
+    seriesType: 'line',
+    services,
+    unit: 'percent',
+    discoverFetch$,
+    searchSessionId,
+    color: chartPalette[6],
+    abortController,
+    chartRef,
+  });
+
+  if (!errorRateLensProps) {
+    return null;
+  }
+
+  return (
+    <ChartContainer size="s" ref={chartRef}>
+      <LensWrapper
+        lensProps={errorRateLensProps}
+        services={services}
+        onBrushEnd={onBrushEnd}
+        onFilter={onFilter}
+        abortController={abortController}
+        syncCursor
+        syncTooltips
+      />
+    </ChartContainer>
```

Comment:
Could we reuse the [Chart](https://github.com/elastic/kibana/blob/5330a0bd88df7dc974f2a9c15eb409b3581a7bdd/src/platform/packages/shared/kbn-unified-metrics-grid/src/components/chart/index.tsx#L39) component - it includes a loading state and a memoized version of `LensWrapperMemo`? Maybe some changes would be needed there